src/JLD.jl

```diff
@@ -953,13 +990,7 @@ end
 function julia_type(e::Union{Symbol, Expr})
     if is_valid_type_ex(e)
         try # `try` needed to catch undefined symbols
-            typ = eval(current_module(), e)
```

Comment:
Removing this is what broke the test, but I fully agree that no one should be relying on this behavior.